hydromt/models/model_api.py

```diff
@@ -115,12 +125,41 @@ def __init__(
         self._root = ""
         self._read = True
         self._write = False
+        self._defered_file_closes = []
 
         # model paths
         self._config_fn = self._CONF if config_fn is None else config_fn
         self.set_root(root, mode)  # also creates hydromt.log file
         self.logger.info(f"Initializing {self._NAME} model from {dist} (v{version}).")
 
+    def _cleanup(self, forceful_cleanup=False, max_close_attempts=2) -> List[str]:
```

Comment:
Could you add a docsting briefly explaining what this method does?
I think I understand, but that's always helpful for future reference

Reply:
Added a docstring, essentially containing my comment above.